src/IECoreDelight/Renderer.cpp

```diff
@@ -1188,6 +1212,28 @@ class DelightRenderer final : public IECoreScenePreview::Renderer
 					m_oversampling = 9;
 				}
 			}
+			else if( name == g_importancesamplefilterOptionName )
+			{
+				if( value )
+				{
+					if( const IntData *d = reportedCast<const IntData>( value, "option", name ) )
+					{
+						if( m_importancesamplefilter != d->readable() )
+						{
+							stop();
+							m_importancesamplefilter = d->readable();
```

Comment:
I don't see why it's necessary to store this in `m_importancesamplefilter` and then wait till `render()` to apply it. Can't we just apply it to the screen node directly here? It seems like the `output()` function is modifying the screen node directly, so I don't see why we shouldn't do the same here.

Reply:
Yes, you are right. Technically it's not necessary to pass the importancesamplefilter value through m_importancesamplefilter to the updateCamera() function inside render(). Since the NSI screen node is being created at the beginning of the Gaffer NSI scene creation, we could set screen attributes like importancesamplefilter and oversampling within the option() function (basically some Gaffer DelightOptions parameters would set NSI global attributes and others would set NSI screen attributes). 

The main benefit of creating the importancesamplefilter and oversampling screen attributes in the render() / updateCamera() function is to have all the screen attributes at the same place in the NSI scene, which makes for a much more readable NSI file. 

The output() function does not set screen attributes, it only connects the outputlayer NSI node to the screen node, so if you prefer to remove importancesamplefilter and oversampling from render() / updateCamera(), I can definitely update option() to have them created there.

Reply:
You bring a really good point on managing code complexity. I spent a bit of time considering the NSI scene readability and actually having the oversampling and importancesamplefilter screen attributes along with the global attributes is not bad at all, since in many ways they are very close to being global attributes (plus Gaffer doesn't currently support multiple screens). I have kept the creation of the oversampling and importancesamplefilter attributes to option() and removed the m_ variables and the respective code in render() / updateCamera(). I also updated the RenderTest.py with a new test for the non-camera screen attributes - testScreenOptions()